### PR TITLE
hf iclass encrypt/decrypt - now use cliparser

### PR DESCRIPTION
**Pull request by tcprst**

### CODE REVIEW

File: client/src/cmdhficlass.c [hunk 1]
```diff
@@ -1049,74 +1010,82 @@ static int CmdHFiClassEView(const char *Cmd) {
     free(dump);
     return PM3_SUCCESS;
 }
-
 static int CmdHFiClassDecrypt(const char *Cmd) {
+    CLIParserContext *clictx;
+    CLIParserInit(&clictx, "hf iclass decrypt",
+                  "3DES decrypt data\n"
+                  "This is naive implementation, it tries to decrypt every block after block 6.\n"
```
Comment (iceman1001) [inline client/src/cmdhficlass.c:1017]: The spelling mistake was before but shouldn't it be  "This is a naive" ?

File: client/src/cmdhficlass.c [hunk 2]
```diff
@@ -1049,74 +1010,82 @@ static int CmdHFiClassEView(const char *Cmd) {
...
+                  "hf iclass decrypt -d 1122334455667788 --key 000102030405060708090a0b0c0d0e0f");
 
-    bool errors = false;
-    bool have_key = false;
-    bool have_data = false;
+    void *argtable[] = {
+        arg_param_begin,
+        arg_str0("f", "file", "<filename>", "filename of dumpfile"),
+        arg_str0("d", "data", "<encrypted blk>", "3DES encrypted data"),
+        arg_str0(NULL, "key", "<transport key>", "3DES transport key"),
```
Comment (iceman1001) [inline client/src/cmdhficlass.c:1030]: keep the   "k"   short param for keys.

Comment (tcprst) [reply client/src/cmdhficlass.c:1030]: we got rid of it in others so I figured we should do the same here.

File: client/src/cmdhficlass.c (hunk 1 above)
Comment (tcprst) [reply client/src/cmdhficlass.c:1017]: Yup, I'll fix that!

File: client/src/cmdhficlass.c (hunk 2 above)
Comment (iceman1001) [reply client/src/cmdhficlass.c:1030]: did we?  in iclass commands?

**Comment by iceman1001** [comment]
Great!

**Pull request merged**